test/infer/test_gradient.py

```diff
@@ -366,3 +366,90 @@ def actual_loss_fn(params_raw):
 
             assert_equal(actual_loss, expected_loss, prec=1e-5)
             assert_equal(actual_grads, expected_grads, prec=1e-5)
+
+
+def kl_model_8_z1z2(params):
+    # Model
+    # z1
+    #
+    # z2
+    probs_z1 = jnp.array([0.5, 0.5])
+    probs_z2 = jnp.array(0.0)
+    pyro.sample("z1", dist.Categorical(probs_z1))
+    pyro.sample("z2", dist.Normal(probs_z2, 1.0))
+
+
+def kl_model_9_z1z2(params):
+    # Model
+    # z1
+    # |
+    # v
+    # z2
+    probs_z1 = jnp.array([0.5, 0.5])
+    probs_z2 = jnp.array([0.0, 0.0])
+    z1 = pyro.sample("z1", dist.Categorical(probs_z1))
+    pyro.sample("z2", dist.Normal(probs_z2[z1], 1.0))
+
+
+@pytest.mark.parametrize(
+    "model",
+    [kl_model_8_z1z2, kl_model_9_z1z2],
+)
+def test_analytic_kl_continuous(model):
```

Comment:
i'm not entirely sure i understand what this testing. can we include a branch with score function gradients for the discrete latient variable, i.e. no enumeration?